File: 012HelloToast/app/src/main/java/com/example/a012hellotoast/MainActivity.java
package com.example.a012hellotoast;

import static androidx.core.content.PackageManagerCompat.LOG_TAG;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.widget.TextView;
import android.widget.Toast;

public class MainActivity extends AppCompatActivity {
    public static final String EXTRA_MESSAGE  =
            "com.example.android.twoactivities.extra.MESSAGE";
    public static final int TEXT_REQUEST = 1;
    private int mCount = 0;
    private TextView mShowCount;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        mShowCount = (TextView) findViewById(R.id.show_count);
    }

    public void showToast(View view) {
        Toast toast = Toast.makeText(this, R.string.toast_message,Toast.LENGTH_SHORT);
        toast.show();
    }

    public void countUp(View view) {
        mCount++;
        if(mShowCount != null){
            mShowCount.setText(Integer.toString(mCount));
        }
    }

    public void launchSecondActivity(View view) {
        Intent intent = new Intent(this, SecondActivity.class);
        mCount++;
        String message = Integer.toString(mCount);//mMessageEditTExt의 텍스트를 message에 삽입
        intent.putExtra(EXTRA_MESSAGE, message);//인텐트에 EXTRA...를 키로 message를 값으로 추가
        startActivityForResult(intent,TEXT_REQUEST);

    }
}

File: 012AnatomyofBasic/app/src/main/java/com/example/a012anatomyofbasic/MainActivity.kt
package com.example.a012anatomyofbasic

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.Button
import android.widget.TextView
import android.widget.Toast

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val rollButton: Button = findViewById(R.id.roll_button)
        rollButton.setOnClickListener { rollDice() }

        val countUpButton: Button = findViewById(R.id.countUp_button)
        countUpButton.setOnClickListener { countUp() }

        val resetButton: Button = findViewById(R.id.reset_button)
        resetButton.setOnClickListener { reset() }



    }

    private fun rollDice() {
        val resultText: TextView = findViewById(R.id.result_text)
        val randomInt = (1..6).random()
//        Toast.makeText(this, "button clicked",
//            Toast.LENGTH_SHORT).show()
        resultText.text = randomInt.toString()

//        resultText.text = "Dice Rolled!"
    }

    private fun countUp(){
        val resultText: TextView = findViewById(R.id.result_text)

        if(resultText.text == "Hello World!") {
            resultText.text = "1"
        }else{
            var resultNum = resultText.text.toString().toInt()

            if(resultNum < 6)
                resultNum++
            resultText.text = resultNum.toString()

        }
    }

    private fun reset(){
        val resultText: TextView = findViewById(R.id.result_text)

        resultText.text = "0"
    }
}



